lib/addons/velero/index.ts

```diff
@@ -0,0 +1,227 @@
+        create: true
+    },
+    values:{
+        initContainers:[
+            {
+                name: "velero-plugin-for-aws",
+                image: "velero/velero-plugin-for-aws:v1.2.0",
+                imagePullPolicy: "IfNotPresent",
+                volumeMounts:[
+                    {
+                        mountPath: "/target",
+                        name: "plugins"
+                    }
+                ]
+            }
+        ],
+        configuration: {
+            provider: "aws",
+            backupStorageLocation:{
+                name: "default",
+                config:{}
+            },
+            volumeSnapshotLocation:{
+                name: "default",
+                config:{}
+            },
+        },
+        serviceAccount: {
+            server:{}
+        }
+    },
+
+};
+
+export class VeleroAddOn implements ClusterAddOn {
+
+    private options: VeleroAddOnProps;
+    constructor(props?: VeleroAddOnProps) {
+        if (props) {
+            // deepmerge the nested json files
+            this.options = deepmerge(defaultProps, props);
+        }
+        else {
+            this.options = defaultProps
+        }
+    }
+
+    deploy(clusterInfo: ClusterInfo): void {
+        const cluster = clusterInfo.cluster;
+        const props = this.options;
+        
+        let bucketName: string; // AWS S3 Bucketname
+        let veleroNamespace: string; // K8s namespace that Velero get deployed onto
+       
+        // Create S3 bucket if no existing bucket, create s3 bucket and corresponding KMS key
+        if ( !props.values.configuration.backupStorageLocation.bucket ){
+             console.log("existing S3 Bucket does not exists, creating S3 bucket");
+             const bucket = new s3.Bucket(cluster, "velero-backup-bucket", {
+                encryption: s3.BucketEncryption.KMS_MANAGED, // Velero Known bug for support with S3 with SSE-KMS with CMK, thus it does not support S3 Bucket Key: https://github.com/vmware-tanzu/helm-charts/issues/83
+                blockPublicAccess: s3.BlockPublicAccess.BLOCK_ALL, // Block Public Access for S3
+                publicReadAccess: false,
+                enforceSSL: true // Encryption in Transit
+            });
+           
+            // Create S3 VPC Endpoint for the Velero pod to access S3 via VPC Endpoint instead of going to internet
+            cluster.vpc.addGatewayEndpoint("velero-backup-bucket-vpcEndPoint", {
+                service: GatewayVpcEndpointAwsService.S3
+            })
+            bucketName = bucket.bucketName;
+        }
+        else {
+            bucketName = props.values.configuration.backupStorageLocation.bucket
+        }
+
+        // Create Namespace if not specified
+        if (props.namespace){
+            // Create Namespace if the "create" option is false
+            if (props.namespace.create) {
+                console.log ("namespace:" + props.namespace.name + " does not existed, creating")
```

Comment:
Let's drop the console.log. If there is any output that you think customer should be aware of, please use `CfnOutput`